async/logic/PostLogic.java

```diff
@@ -0,0 +1,345 @@
+package com.open.jp.async.logic;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+import java.time.LocalDateTime;
+import java.util.Map;
+import java.util.Properties;
+import java.util.UUID;
+
+import javax.enterprise.context.RequestScoped;
+import javax.ws.rs.core.Response;
+
+import com.open.jp.async.errorresponse.ErrorResponse;
+import com.open.jp.async.request.Request;
+import com.open.jp.async.response.Responses;
+import com.open.jp.async.util.StringUtil;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.serialization.StringSerializer;
+
+
+@RequestScoped
+public class PostLogic {
+	
+	
+	private StringUtil stringutil = new StringUtil();
+	
+	public PostLogic() {
+		
+	}
+	
+		/**
+	 * Description<br>
+	 * エラー時のレスポンスを生成するメソッド.<br>
+	 * 
+	 * ErrorResponseEnumで定義されている、ErrorStatusCodeMessage、ErrorMessage、ErrorStatusをErrorResponseクラスの
+	 * Time、Code、Messageにセットし各エラーステータスコードによって呼び出す際に利用する.<br>
+	 * 
+	 * @version 2.0
+	 * @param errorresponseenum "E400" "E409" "E500" ErrorResponseEnumで定義されたエラーコードを引数とする.
+	 * @return エラーの場合は"ErrorResponseEnum"クラスで定義されている各エラーとそのメッセージを返す。
+	 */
+	public  Response GeneralErrorRes(ErrorResponseEnum errorresponseenum) {
+	    ErrorResponse erroresponse = new ErrorResponse();
+	    erroresponse.setTime(LocalDateTime.now().toString()); 
+	    erroresponse.setCode(errorresponseenum.getErrorStatusCodeMessage()); 
+	    erroresponse.setMessage(errorresponseenum.getErrorMessage()); 
+	    return Response.status(errorresponseenum.getErrorStatus()).entity(erroresponse).build();
+	}
+	
+	public Response service(Request request, Map<String,Object> body) {
+		
+		Responses response = new Responses();
+		StringBuilder asyncID = new StringBuilder();
+		StringBuilder inputStrings = new StringBuilder();
+		
+		final String NORMAL_END = "Successfull";
+		
+		Response typeresponse = headerTypeCheck(request);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *typeresponse*. The resource is referenced at line 63. The resource is returned at line 64. There are other execution paths that don&#39;t close the resource or return it, for example, when an else condition at line 63 evaluates to true. Ensure *typeresponse* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).